frontend/app/src/components/Countdown/Countdown.tsx

```diff
@@ -14,58 +14,40 @@
  * limitations under the License.
  */
 
-import React, { PureComponent, ReactNode } from "react"
+import React, { useState } from "react"
 
 import { StyledCountdown } from "./styled-components"
 
 interface Props {
   countdown: number
-  endCallback: () => void
+  endCallback?: () => void
 }
 
-interface State {
-  countdown: number
-}
-
-class Countdown extends PureComponent<Props, State> {
-  public static defaultProps: Partial<Props> = {
-    endCallback: () => {},
-  }
+const Countdown: React.FC<Props> = ({ countdown, endCallback = () => {} }) => {
```

Comment:
nit: instead of the defining a default function that does not do anything, we could just have it `undefined` and check for it [here](https://github.com/streamlit/streamlit/pull/9734/files#diff-e35fe54ffc61c545130b65ee4d5c54021587435201f5aa7eea426153dbd182e7R36)

Reply:
I made it a required prop again which should solve this as well.